.devcontainer/images/Dockerfile

```
@@ -232,6 +232,14 @@ USER vscode
 # Provides: coderabbit (cr) - local AI code reviews
 RUN curl -fsSL https://cli.coderabbit.ai/install.sh | sh
 
+# =============================================================================
+# Qodo Command CLI (AI-powered code review + agents)
+# =============================================================================
+# Provides: qodo - local AI code reviews, chains, custom agents
+# Docs: https://docs.qodo.ai/qodo-documentation/qodo-command
+# Note: npm available via Claude Code's bundled Node.js runtime
+RUN npm install -g @qodo/command
```

Hadolint (2.14.0)

[warning] Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`

(DL3016)